mlflow/azureml/__init__.py

```diff
@@ -206,8 +207,163 @@ def build_image(model_uri, workspace, image_name=None, model_name=None,
+                              `
+    :param mlflow_home: Path to a local copy of the MLflow GitHub repository. If specified, the
+                        image will install MLflow from this directory. Otherwise, it will install
+                        MLflow from pip.
+    :param synchronous: If ``True``, this method blocks until the image creation procedure
+                        terminates before returning. If ``False``, the method returns immediately,
+                        but the returned image will not be available until the asynchronous
+                        creation process completes. Use the
+                        ``azureml.core.Image.wait_for_creation()`` function to wait for the creation
+                        process to complete.
+    :return: A tuple containing the following elements in order:
+            - An ``azureml.core.image.ContainerImage`` object containing metadata for the new image.
+            - An ``azureml.core.model.Model`` object containing metadata for the new model.
+
+    >>> import mlflow.azureml
+    >>> from azureml.core import Workspace
+    >>> from azureml.core.webservice import AciWebservice, Webservice
+    >>>
+    >>> # Load or create an Azure ML Workspace
+    >>> workspace_name = "<Name of your Azure ML workspace>"
+    >>> subscription_id = "<Your Azure subscription ID>"
+    >>> resource_group = "<Name of the Azure resource group in which to create Azure ML resources>"
+    >>> location = "<Name of the Azure location (region) in which to create Azure ML resources>"
+    >>> azure_workspace = Workspace.create(name=workspace_name,
+    >>>                                    subscription_id=subscription_id,
+    >>>                                    resource_group=resource_group,
+    >>>                                    location=location,
+    >>>                                    create_resource_group=True,
+    >>>                                    exist_okay=True)
+    >>>
+    >>> # Build an Azure ML Container Image for an MLflow model
+    >>> azure_service, azure_model = mlflow.azureml.deploy(
+    >>>                                 model_uri="<model_uri>",
+    >>>                                 service_name="<deployment-name>"
+    >>>                                 workspace=azure_workspace,
+    >>>                                 synchronous=True)
+    >>>
+    """
+    # The Azure ML SDK is only compatible with Python 3. However, the `mlflow.azureml` module should
+    # still be accessible for import from Python 2. Therefore, we will only import from the SDK
+    # upon method invocation.
+    # pylint: disable=import-error
+    from azureml.core.model import Model as AzureModel, InferenceConfig
+    from azureml.core import Environment as AzureEnvironment
+    from azureml.core import VERSION as AZUREML_VERSION
+
+    absolute_model_path = _download_artifact_from_uri(model_uri)
+
+    model_pyfunc_conf, run_id = _load_pyfunc_conf(model_path=absolute_model_path)
```

Comment:
As discussed in `_load_pyfunc_conf`, a run ID is not always guaranteed to exist in the MLmodel configuration. For example, a user may choose to invoke `mlflow.pyfunc.save_model` to produce a model independently of an MLflow run. This seems to impact the `model_name` and `service_name` generated below. Are there other naming schemes that you could adopt without relying on run ID?

Reply:
We should also make sure to test the case where models without run IDs are deployed via this method.

Reply:
I'm going with a Guid if run ID isn't present but perhaps a timestamp would be more useful? not really sure.